polxml

```diff
@@ -65,52 +65,128 @@ def intfromhexiflarge(a):
 
 # --------------------------------------
 
+atomic_wt = {'H': 1.008, 'Li': 6.941, 'B': 10.811, 'C': 12.011,
+             'N': 14.006, 'O': 15.999, 'F': 18.998, 'Ne': 20.180,
+             'Na': 22.990, 'Mg': 24.305, 'Al': 26.982, 'Si':  28.086,
+             'P': 30.974, 'S': 32.065, 'Cl': 35.453, 'Ar': 39.948,
+             'K': 39.098, 'Ca': 40.078, 'Ti': 47.867, 'Fe': 55.845,
+             'Cu': 63.546, 'Zn': 65.38, 'Se': 78.971, 'Br': 79.904, 'Kr': 83.798,
+             'Mo': 95.96, 'Ru': 101.07, 'Sn': 118.710, 'Te': 127.60,
+             'I': 126.904, 'Xe': 131.293}
+
+def atomic_symbol(name):
+    if name[:2] in atomic_wt:
+        return name[:2]
+    elif name[0] in atomic_wt:
+        return name[0]
+    else:
+        print('warning: unknown symbol for atom ' + name)
+        return name
+
+# --------------------------------------------
+
 class Topology(object):
-    '''topology and coordinates from pdb file'''
+    '''topology and coordinates from pdbx/mmcif file'''
 
     def __init__(self, pdbfile):
-        '''read from pdb file'''
+        '''read from pdbx file'''
 
-        self.box = ''
+        self.box = []
         self.atoms = []
         self.bonds = []
         self.natom = 0
         self.ndrude = 0
-
-        with open(pdbfile, 'r') as f:
-            line = f.readline()
-            while 'CRYST1' not in line:
-                line = f.readline()
-            self.box = line.strip()
-            line = f.readline()
-            while not (line.startswith('ATOM') or line.startswith('HETATM')):
+        
+        if pdbfile == 'config.pdb':
```

Comment:
Maybe this should go into 2 different functions, one to read the pdf and the other to read pdbx, rather than in an if statement. First test which file type then readpdb() or readpdbx()